bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -255,10 +255,15 @@ func (r *KubeadmConfigReconciler) handleClusterNotInitialized(ctx context.Contex
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	machine := &clusterv1.Machine{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(scope.ConfigOwner.Object, machine); err != nil {
```

Comment:
Would it be good to do a check that `scope.ConfigOwner.GetKind() == "Machine"` here first?

Reply:
Yes, for now I'll restrict control plane related operations to Machine since we don't plan on supporting MachinePool control planes in the immediate future.